src/main/java/knightminer/inspirations/recipes/block/BlockSmashingAnvil.java

```diff
@@ -21,11 +34,115 @@ public BlockSmashingAnvil() {
 	@Override
 	public void onEndFalling(World world, BlockPos pos, IBlockState anvil, IBlockState state) {
 		BlockPos down = pos.down();
-		if(!smashBlock(world, down, world.getBlockState(down))) {
+		IBlockState blockState = world.getBlockState(down);
+		// smash the items and then the block and do both in combination
+		if(!smashItem(world, down.up(), blockState) & !smashBlock(world, down, blockState)) {
 			super.onEndFalling(world, pos, anvil, state);
 		}
 	}
 
+	/**
+	 * Smash an item stack in the world with an anvil
+	 * @param world the world
+	 * @param pos the position the anvil landed on
+	 * @param state the state of the block the anvil landed on
+	 * @return true if the item was smashed
+	 */
+	public static boolean smashItem(World world, BlockPos pos, IBlockState state) {
+		// find item entities that can be smashed
+		List<EntityItem> entityItem = getItemEntity(world, pos);
+		if(entityItem.isEmpty()) {
+			return false;
+		}
+
+		NonNullList<ItemStack> inputs = NonNullList.create();
+		inputs.addAll(entityItem.stream().map(EntityItem::getItem).collect(Collectors.toList()));
+
+		int fallHeight = getFallHeight(world, pos);
+
+		// repeat as long as a recipe matches
+		List<ItemStack> results = new ArrayList<>();
+		boolean recipeApplied = false;
+		int iterations = 0;
+		while(iterations++ < MAX_RECIPE_APPLICATIONS) {
```

Comment:
I would rather not be scanning the recipe registry many times in a single landing. Ideally just one recipe would catch then we reuse it until we have our results, so if we have too many inputs for a recipe it just does nothing.

It might also be a good idea to have the recipe just handle stacks using math instead of a loop. So for somethimg like the bone example, instead of saying the result is bonemeal * 4, its bonemeal * 4 * inputStackSize.